File: src/main/java/io/github/cdgeass/editor/dom/element/Statement.java
package io.github.cdgeass.editor.dom.element;

import com.intellij.psi.PsiMethod;
import com.intellij.util.xml.Convert;
import com.intellij.util.xml.DomElement;
import com.intellij.util.xml.GenericAttributeValue;
import io.github.cdgeass.editor.dom.convert.PsiMethodReferenceConvert;

/**
 * @author cdgeass
 * @since 2020-05-26
 */
public interface Statement extends DomElement {

    @Convert(PsiMethodReferenceConvert.class)
    GenericAttributeValue<PsiMethod> getId();
}

File: src/main/kotlin/io/github/cdgeass/codeInsight/dom/element/Statement.kt
package io.github.cdgeass.codeInsight.dom.element

import com.intellij.psi.PsiMethod
import com.intellij.util.xml.Convert
import com.intellij.util.xml.DomElement
import com.intellij.util.xml.GenericAttributeValue
import com.intellij.util.xml.NameValue
import com.intellij.util.xml.Required
import io.github.cdgeass.codeInsight.dom.converter.MyPsiMethodConverter

/**
 * @author cdgeass
 * @since 2021/3/19
 */
interface Statement : DomElement {
    /**
     * Returns the value of the id child.
     * Attribute id
     * @return the value of the id child.
     */
    @Required
    @NameValue
    @Convert(MyPsiMethodConverter::class)
    fun getId(): GenericAttributeValue<PsiMethod>
}
